client/coral-framework/hocs/withReaction.js

```diff
@@ -0,0 +1,238 @@
+import React from 'react';
+import get from 'lodash/get';
+import uuid from 'uuid/v4';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import {compose, gql, graphql} from 'react-apollo';
+import withFragments from 'coral-framework/hocs/withFragments';
+import {showSignInDialog} from 'coral-framework/actions/auth';
+import {capitalize} from 'coral-framework/helpers/strings';
+import {getMyActionSummary, getTotalActionCount} from 'coral-framework/utils';
+
+export default reaction => WrappedComponent => {
+  if (typeof reaction !== 'string') {
+    console.error('Reaction must be a valid string');
+    return null;
+  }
+
+  reaction = reaction.toLowerCase();
+
+  class WithReactions extends React.Component {
+    render() {
+      const {comment} = this.props;
+
+      const reactionSummary = getMyActionSummary(
+        `${capitalize(reaction)}ActionSummary`,
+        comment
+      );
+
+      const count = getTotalActionCount(
+        `${capitalize(reaction)}ActionSummary`,
+        comment
+      );
+
+      const withReactionProps = {reactionSummary, count};
+
+      return <WrappedComponent {...this.props} {...withReactionProps} />;
```

Comment:
@jde What you mean is the display name of the resulting component and this needs to be defined in the `displayName` static variable, see our other HOCs that we already have in the framework.